File: generators/app/templates/java/app/src/main/java/org/onepointzero/androidstarterjava/application/MainPresenter.java
package <%= packageName %>.application;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.disposables.Disposables;
import io.reactivex.schedulers.Schedulers;

public class MainPresenter {
    private final GetStarterData getStarterData;
    private MainScreen screen;

    private List<Disposable> disposables = new ArrayList<>();

    public MainPresenter(GetStarterData getStarterData) {
        this.getStarterData = getStarterData;
    }

    void bind(MainScreen screen) {
        this.screen = screen;

        Disposable disposable = getStarterData.execute()

                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())

                .subscribe(message -> screen.showLabel(message));

        disposables.add(disposable);
    }

    void unbind() {
        for (Disposable disposable: disposables) {
            disposable.dispose();
        }
    }
}

File: generators/app/templates/kotlin/app/src/main/java/org/onepointzero/androidstarterkotlin/application/MainPresenter.kt
package <%= packageName %>.application

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers


class MainPresenter(private val getStarterData: GetStarterData) {

    fun bind(screen: MainScreen) {
        getStarterData.execute()

                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())

                .subscribe {
                    item -> screen.printText(item)
                }

    }
}